dbms/src/Storages/MergeTree/IMergeTreeDataPartWriter.h

```diff
@@ -0,0 +1,164 @@
+#pragma once
+
+#include <IO/WriteBufferFromFile.h>
+#include <IO/WriteBufferFromFileBase.h>
+#include <Compression/CompressedWriteBuffer.h>
+#include <IO/HashingWriteBuffer.h>
+#include <Storages/MergeTree/MergeTreeData.h>
+#include <DataStreams/IBlockOutputStream.h>
+#include <Storages/MergeTree/IMergeTreeDataPart.h>
+
+
+namespace DB
+{
+
+
+/// Writes data part to disk in different formats.
+/// Calculates and serializes primary and skip indices if needed.
+class IMergeTreeDataPartWriter : private boost::noncopyable
+{
+public:
+    using WrittenOffsetColumns = std::set<std::string>;
+
+    struct ColumnStream
```

Comment:
Comment. Why we have this class?

Reply:
I also renamed this class, because it can write not only column.